src/ray/core_worker/core_worker.h

```diff
@@ -981,6 +986,8 @@ class CoreWorker : public rpc::CoreWorkerServiceHandler {
   /// Manages recovery of objects stored in remote plasma nodes.
   std::unique_ptr<ObjectRecoveryManager> object_recovery_manager_;
 
+  // TODO(swang): Refactor to merge actor_handles_mutex_ and all fields that it
```

Comment:
that would be great! thought of doing this while implementing detached actors too...

Reply:
I did it in my next PR for actor restart :)